ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -983,6 +988,19 @@ private class JdbcLedgerDao(
       }
   }
 
+  // The old indexer (com.daml.platform.indexer.JdbcIndexer) uses IncrementalOffsetStep,
+  // and we have tests in JdbcLedgerDao*Spec checking that this is handled correctly.
+  // The append-only schema and the parallel ingestion indexer doesn't use this class.
+  // TODO append-only: Remove IncrementalOffsetStep along with this method and all corresponding tests.
```

Comment:
Do we have a reason to keep the OffsetStep at all? If not I would change this TODO, to let it be the TODO for that.
(It seems to me the changes here changing the LedgerDao methods Offset -> OffsetStep needs to be reverted upon cleanup...was that your intention?)

Reply:
Good point, not only can we remove `IncrementalOffsetStep`, we can remove the whole `OffsetStep` trait. I'll change the TODO.

The changes around `Offset -> OffsetStep` were only required so that the append-only JdbcLedgerDao can use the existing `JdbcLedgerDao*Spec` tests.